app/src/main/java/org/fossasia/susi/ai/activities/MainActivity.java

```diff
@@ -328,7 +328,8 @@ private void parseSusiResponse(SusiResponse susiResponse) {
 
     private void getOldMessages() {
         if (isNetworkConnected()) {
-            Call<MemoryResponse> call = clientBuilder.getSusiApi().getChatHistory();
+            ClientBuilder clientBuilder1 = new ClientBuilder();
+            Call<MemoryResponse> call = clientBuilder1.getSusiApi().getChatHistory();
```

Comment:
Exception was there since you called retrieveOldMessage before initializing clientBuilder. Initialize it above call to retrieveOldMessages() in onCreate ()